src/cryptography/hazmat/primitives/keywrap.py

```diff
@@ -68,6 +68,63 @@ def _unwrap_core(wrapping_key, a, r, backend):
     return a, r
 
 
+def aes_key_wrap_with_padding(wrapping_key, key_to_wrap, backend):
+    if len(wrapping_key) not in [16, 24, 32]:
+        raise ValueError("The wrapping key must be a valid AES key length")
+
+    aiv = b"\xA6\x59\x59\xA6" + struct.pack(">i", len(key_to_wrap))
+    r = [key_to_wrap[i:i + 8] for i in range(0, len(key_to_wrap), 8)]
+    pad = (8 - (len(r[-1]) % 8)) % 8
+    # Pad the last block if necessary
+    r[-1] = r[-1] + b"\x00" * pad
+    if len(r) == 1:
+        # RFC 5649 - 4.1 - exactly 8 octets after padding
+        encryptor = Cipher(AES(wrapping_key), ECB(), backend).encryptor()
+        b = encryptor.update(aiv + r[0])
+        assert encryptor.finalize() == b""
+        return b
+    else:
+        return _wrap_core(wrapping_key, aiv, r, backend)
+
+
+def aes_key_unwrap_with_padding(wrapping_key, wrapped_key, backend):
+    if len(wrapped_key) < 16:
```

Comment:
Is this necessary since we assert the wrapping key is in [16, 24, 32] just below?

We don't do this length < 16 check for the wrapping function, so we should either do it for both wrap/unwrap or neither